text/0000-explicit-tail-calls.md

```diff
@@ -249,31 +245,23 @@ fn bar(n: i32) {
 - Corner cases are dissected by example.
 
 The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work. -->
-This explanation is mostly based on the [previous RFC](https://github.com/DemiMarie/rfcs/blob/become/0000-proper-tail-calls.md#detailed-design)
-though is more restricted as the current RFC does not target general tail calls anymore.
+Implementation of this feature requires checks that all prerequisites to guarantee TCE are fulfilled.
+These checks are:
+- The `become` keyword is only used in place of `return`. The intend is to reuse the semantics of a `return` signifying "the end of a function". See the section on [tail-call-elimination](#tail-call-elimination) for examples.
+- The argument to `become` is a function (or method) call, that exactly matches the function signature and calling convention of the callee. The intend is to assure a compatible stack frame layout.
+- The stack frame of the calling function is reused, this also implies that the function is never returned to. The required checks to ensure this is possible are: no local variables are passed to the called function, and no further cleanup is necessary. These checks can be done by using the borrowchecker as described in the [Borrowchecking](#semantics) section below.
```

Comment:
```suggestion
- The stack frame of the calling function is reused, this also implies that the function is never returned to. The required checks to ensure this is possible are: no borrows of local variables are passed to the called function (passing local variables by copy/move is ok, since that doesn't require the local variable to continue existing after the call), and no further cleanup is necessary. These checks can be done by using the borrowchecker as described in the [Borrowchecking](#semantics) section below.
```

Reply:
Ah yes, I forgot to mention that part. Thanks for taking another look!